service/src/utils.rs

```diff
@@ -45,12 +47,13 @@ pub fn hex_encode(data: Vec<u8>) -> String {
 	hex_str
 }
 
-pub fn write_slice_and_whitespace_pad(writable: &mut [u8], data: Vec<u8>) {
-	assert!(!data.len() > writable.len(), "Not enough bytes in output buffer for return value");
+pub fn write_slice_and_whitespace_pad(writable: &mut [u8], data: Vec<u8>) -> Result<(), Error> {
+	ensure!(data.len() <= writable.len(), Error::InsufficientBufferSize);
 	let (left, right) = writable.split_at_mut(data.len());
 	left.clone_from_slice(&data);
 	// fill the right side with whitespace
 	right.iter_mut().for_each(|x| *x = 0x20);
+	Ok(())
```

Comment:
Changed the signature of this method to return a result in case of failure (instead of panicking). Otherwise this could lead to unintentional shut downs of a worker.